### PR TITLE
ES|QL: register TABLES_TYPES capability and fix CSV tests

### PR DESCRIPTION
**Pull request by luigidellaquila**
Registering `TABLES_TYPES` capability, that was created (but not registered) in https://github.com/elastic/elasticsearch/pull/109489

Also using it as a condition for `lookup.csv-spec` tests.
These tests failed badly in Serverless multi-cluster due to different output format in two different nodes.

Probably (if I got it right) the YAML tests that were using this capability were not running at all. This PR should re-enable them.

### CODE REVIEW

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java [hunk 1]
```diff
@@ -92,6 +92,7 @@ private static Set<String> capabilities() {
 
         if (Build.current().isSnapshot()) {
             caps.add(LOOKUP_COMMAND);
+            caps.add(TABLES_TYPES);
```
Comment (alex-spies) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java:95]: nit: I'd argue that we should actually remove `LOOKUP_COMMAND`, as `TABLES_TYPES` introduced a breaking change and should therefore be treated as a `LOOKUP_COMMAND_V2`.

If we do this, we should align the yaml tests as well - these currently require both `tables_types` AND `lookup_command` and would be skipped if we removed `LOOKUP_COMMAND`.

For our test suites, it should make no difference if we have both caps here or only `TABLES_TYPES`; however, having both capabilities is like a node was saying "I can handle both the old and the new lookup!", which is wrong and could lead to failures in CCQ-scenarios where the remote cluster is _newer_ than the local cluster.

**Review by alex-spies — APPROVED** [review]
Thanks Luigi!

**Review by nik9000 — APPROVED** [review]
Yeah, it does make sense to swap it to the new one. It's nice to have the flexibility to just do whatever we need to do with these capabilities.

**Comment by luigidellaquila** [comment]
Thanks for the feedback folks!

**Comment by luigidellaquila** [comment]
@elasticmachine update branch

**Comment by luigidellaquila** [comment]
@elasticmachine update branch

**Comment by luigidellaquila** [comment]
@elasticmachine update branch

**Comment by luigidellaquila** [comment]
@elasticmachine update branch

**Comment by luigidellaquila** [comment]
@elasticmachine update branch

**Comment by luigidellaquila** [comment]
@elasticmachine update branch

**Pull request merged**